skimage/exposure/exposure.py

```diff
@@ -120,10 +125,27 @@ def histogram(image, nbins=256, source_range='image', normalize=False):
     (array([107432, 154712]), array([ 0.25,  0.75]))
     """
     sh = image.shape
-    if len(sh) == 3 and sh[-1] < 4:
+    if len(sh) == 3 and sh[-1] < 4 and not multichannel:
         warn("This might be a color image. The histogram will be "
              "computed on the flattened image. You can instead "
-             "apply this function to each color channel.")
+             "apply this function to each color channel, or set "
+             "multichannel=True")
+
+    if multichannel:
+        channels = sh[-1]
+        hist = []
+        bin_centers = []
+        for chan in range(channels):
+            h, bc = _histogram(image[..., chan], nbins, source_range, normalize)
+            hist.append(h)
```

Comment:
since the bins are the same, hist could be transformed into an array before being returned.

Reply:
You're right. I'll fix that.

Reply:
This is not actually correct: if the different channels have different data ranges, and the source range is image, then each histogram will have different bins.

Reply:
This was resolved by passing `bins` instead of `nbins` in ec7edf36ef467d3a398673b4000060d724566d33